common/expbloom/bloom_test.go

```diff
@@ -0,0 +1,85 @@
+package bloom
+
+import (
+	"encoding/binary"
+	"testing"
+	"time"
+)
+
+type hashable []byte
+
+func (h hashable) Write(p []byte) (n int, err error) { panic("not implemented") }
+func (h hashable) Sum(b []byte) []byte               { panic("not implemented") }
+func (h hashable) Reset()                            { panic("not implemented") }
+func (h hashable) BlockSize() int                    { panic("not implemented") }
+func (h hashable) Size() int                         { return 8 }
+func (h hashable) Sum64() uint64 {
+	hash := make([]byte, 8)
+	copy(hash, h)
+	return binary.BigEndian.Uint64(hash[0:8])
+}
+
+func TestBloom(t *testing.T) {
+	bloom, _ := NewExpiringBloom(3, 1024, 10*time.Millisecond)
+
+	testKey := hashable([]byte{0x01})
+	bloom.Add(testKey)
+	if !bloom.Contains(testKey) {
+		t.Fatal()
+	}
+	time.Sleep(11 * time.Millisecond)
```

Comment:
Ouch, yeah, no this is not the kind of code that works well on our CI pipelines :)